x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHashTilerTests.java

```diff
@@ -83,6 +83,14 @@ protected int expectedBuckets(GeoShapeValues.GeoShapeValue geoValue, int precisi
         if (precision == 0) {
             return 1;
         }
+        GeoShapeValues.BoundingBox bounds = geoValue.boundingBox();
```

Comment:
I think this test can be structured better if you generated a random boolean to choose between intentionally generating a bounding box resolving into point or a bounding box that wouldn't resolve into point and then  based on this varian followed one branch or another. Otherwise your test and production is basically relying on the same code.

Reply:
The problem is that both branches might not get the same result for the same point when the point is in one edge of the hash. Geohash#stringEncode might give you one hash but when you iterate using the resolved rectangles you might end up in a different one.

so yes, I am aware is almost the same code but I am not sure we can do better at the moment.

Reply:
Yeah, now that i am thinking about it, it is not enough to just make it small it needs to be small in a very particular way. Oh well. I guess we will have to live with it for now.